core/src/com/unciv/logic/map/MapUnit.kt

```diff
@@ -879,30 +879,40 @@ class MapUnit {
         return damageFactor
     }
 
-    private fun getTerrainDamage() {
-        // hard coded mountain damage for now
-        if (getTile().baseTerrain == Constants.mountain) {
-            val tileDamage = 50
-            health -= tileDamage
+    private fun doTerrainDamage() {
+        val tileDamage = getDamageFromTerrain()
+        health -= tileDamage
 
-            if (health <= 0) {
-                civInfo.addNotification(
-                    "Our [$name] took [$tileDamage] tile damage and was destroyed",
-                    currentTile.position,
-                    name,
-                    NotificationIcon.Death
-                )
-                destroy()
-            } else civInfo.addNotification(
-                "Our [$name] took [$tileDamage] tile damage",
+        if (health <= 0) {
+            civInfo.addNotification(
+                "Our [$name] took [$tileDamage] tile damage and was destroyed",
                 currentTile.position,
-                name
+                name,
+                NotificationIcon.Death
             )
+            destroy()
+        } else if (tileDamage > 0) civInfo.addNotification(
+            "Our [$name] took [$tileDamage] tile damage",
+            currentTile.position,
+            name
+        )
+    }
+
+    fun getDamageFromTerrain(terrainName: String = currentTile.baseTerrain): Int {
+        var totalDamage = 0
+        if (terrainName == Constants.mountain)
+            totalDamage += 50   // The norm
+        if (civInfo.nonStandardTerrainDamage) {
+            for (unique in getMatchingUniques("Units ending their turn on a [] take [] damage")) {
```

Comment:
The way the unique is phrased, it sounds like "X damage" and not "X extra damage", so totalDamage =  and not totalDamage +=, no?